sqlmesh/core/snapshot/evaluator.py

```diff
@@ -1713,6 +1763,9 @@ def promote(
             view_properties=model.render_virtual_properties(**render_kwargs),
         )
 
+        # Apply grants to the virtual layer view
+        self._apply_grants(model, view_name, GrantsTargetLayer.VIRTUAL)
```

Comment:
If all evaluation strategies share this, I suggest calling this in the `SnapshotEvaluator` itself rather than in individual strategies.